packages/block-editor/src/components/link-control/index.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { isFunction, noop, startsWith } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	ExternalLink,
+	Popover,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+import {
+	useCallback,
+	useState,
+	useEffect,
+	Fragment,
+} from '@wordpress/element';
+
+import {
+	safeDecodeURI,
+	filterURLForDisplay,
+	isURL,
+	prependHTTP,
+	getProtocol,
+} from '@wordpress/url';
+
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import LinkControlSettingsDrawer from './settings-drawer';
+import LinkControlSearchItem from './search-item';
+import LinkControlInputSearch from './input-search';
+
+function LinkControl( {
+	instanceId,
+	currentLink,
+	className,
+	fetchSearchSuggestions,
+	onLinkChange,
+	currentSettings,
+	onSettingsChange = { noop },
+	onKeyDown = noop,
+	onKeyPress = noop,
+	onClose = noop,
+} ) {
+	// State
+	const [ inputValue, setInputValue ] = useState( '' );
+	const [ isEditingLink, setIsEditingLink ] = useState( false );
+
+	// Effects
+	useEffect( () => {
+		// If we have a link then stop editing mode
+		if ( currentLink ) {
+			setIsEditingLink( false );
+		} else {
+			setIsEditingLink( true );
+		}
+	}, [ currentLink, setIsEditingLink ] );
```

Comment:
Not sure is `setIsEditingLink` makes sense here, Dave.

Reply:
Actually you are right.

Reply:
Fixed.